src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs

```diff
@@ -90,19 +90,26 @@ protected override async Task<object> InvokeCore(object[] parameters, FunctionIn
 
         private async Task DelayUntilFunctionDispatcherInitializedOrShutdown()
         {
-            if (_functionDispatcher != null && _functionDispatcher.State == FunctionInvocationDispatcherState.Initializing)
+            // Don't delay if functionDispatcher is already initialized OR is skipping initialization for one of
+            // these reasons: started in placeholder, has no functions, functions do not match set language.
+            if (_functionDispatcher == null
+                || _functionDispatcher.State == FunctionInvocationDispatcherState.Initialized
+                || _functionDispatcher.State == FunctionInvocationDispatcherState.Default)
             {
```

Comment:
FunctionInvocationDispatcher is created with `FunctionInvocationDispatcherState.Default`. How about looking specifically for  ```if (_functionDispatcher != null && (_functionDispatcher.State == FunctionInvocationDispatcherState.Initializing || FunctionInvocationDispatcherState.WorkerProcessRestarting)```

Reply:
Another state where we would delay is "Disposing" - would prefer to keep this as skip it if it's Initialized or Default!

Reply:
^wrote this logic a bit wrong though, rewriting!